src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/subjectaccessrequests/SubjectAccessRequestService.kt

```diff
@@ -0,0 +1,167 @@
+package uk.gov.justice.digital.hmpps.approvedpremisesapi.service.subjectaccessrequests
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import org.json.JSONObject
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.api.model.ServiceName
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.CAS1SubjectAccessRequestRepository
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.CAS2SubjectAccessRequestRepository
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.CAS3SubjectAccessRequestRepository
+import uk.gov.justice.hmpps.kotlin.sar.HmppsPrisonProbationSubjectAccessRequestService
+import uk.gov.justice.hmpps.kotlin.sar.HmppsSubjectAccessRequestContent
+import java.time.LocalDate
+import java.time.LocalDateTime
+import java.time.LocalTime
+
+@Service
+class SubjectAccessRequestService(
+  val objectMapper: ObjectMapper,
+  val cas1SubjectAccessRequestRepository: CAS1SubjectAccessRequestRepository,
+  val cas2SubjectAccessRequestRepository: CAS2SubjectAccessRequestRepository,
+  val cas3SubjectAccessRequestRepository: CAS3SubjectAccessRequestRepository,
+) : HmppsPrisonProbationSubjectAccessRequestService {
+
+  private val log = LoggerFactory.getLogger(this::class.java)
+
+  fun getCAS1Result(crn: String?, nomsNumber: String?, startDate: LocalDateTime?, endDate: LocalDateTime?): String {
+    val approvedPremisesApplicationsJson = cas1SubjectAccessRequestRepository.getApprovedPremisesApplicationsJson(crn, nomsNumber, startDate, endDate)
```

Comment:
I think we pass the same 4 arguments to every repository function. I propose we introduce a data class to encapsulate these arguments to reduce repeated code. Can then update getSarResult function to take in this dataclass and pass it through to all functions

Reply:
good call - i'll look into this on the next pass.

Reply:
proposing to leave this now as it affects quite a bit of test code. will look to add a ticket to improve as a tech debt item.